index.html

```diff
@@ -35,58 +41,176 @@
+                connection parameters (or their defaults) specified in |options|.
+              <li>
+                If this fails for any reason, [=queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ol>
+                  <li>
+                    [=Reject=] the {{TCPSocket/[[openedPromise]]}} with a "{{NetworkError}}" {{DOMException}}.
+                  <li>
+                    [=Reject=] the {{TCPSocket/[[closedPromise]]}} with a "{{NetworkError}}" {{DOMException}}.
+                </ol>
+              <li>
+                On success, [=queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ol>
+                  <li>
+                    [=initialize tcp readable stream|Initialize=] {{TCPSocket/[[readable]]}}.
+                  <li>
+                    [=initialize tcp writable stream|Initialize=] {{TCPSocket/[[writable]]}}.
+                  <li>
+                    Let |openInfo:TCPSocketOpenInfo| be a new {{TCPSocketOpenInfo}}.
+                  <li>
+                    Set |openInfo|["{{TCPSocketOpenInfo/readable}}"] to [=this=].{{TCPSocket/[[readable]]}}.
+                  <li>
+                    Set |openInfo|["{{TCPSocketOpenInfo/writable}}"] to [=this=].{{TCPSocket/[[writable]]}}.
+                  <li>
+                    Populate the remaining fields of |openInfo| using the information provided by the operating system:
+                      |openInfo|["{{TCPSocketOpenInfo/remoteAddress}}"], |openInfo|["{{TCPSocketOpenInfo/remotePort}}"],
+                      |openInfo|["{{TCPSocketOpenInfo/localAddress}}"] and |openInfo|["{{TCPSocketOpenInfo/localPort}}"].
+                  <li>
+                    [=Resolve=] [=this=].{{TCPSocket/[[openedPromise]]}} with |openInfo|.
+                </ol>
+            </ol>
+        </ol>
+
+        <section data-dfn-for="TCPSocketOptions">
+          <h4><dfn>TCPSocketOptions</dfn> dictionary</h4>
+
           <pre class="idl">
-          dictionary SocketOptions {
-            DOMString localAddress;
-            unsigned short localPort;
+            dictionary TCPSocketOptions {
+              [EnforceRange] unsigned long sendBufferSize;
+              [EnforceRange] unsigned long receiveBufferSize;
 
-            DOMString remoteAddress;
-            unsigned short remotePort;
+              <!--
+              DOMString localAddress;
+              [EnforceRange] unsigned short localPort;
+              -->
 
-            unsigned long sendBufferSize;
-            unsigned long receiveBufferSize;
-          };
+              boolean noDelay = false;
+              boolean keepAlive = false;
```

Comment:
Since `keepAlive` now defaults to false we can merge it with `keepAliveDelay`.

Reply:
Fixed!